packages/react-native-executorch/common/rnexecutorch/models/speech_to_text/asr/ASR.cpp

```diff
@@ -0,0 +1,302 @@
+#include <random>
+
+#include "ASR.h"
+#include "executorch/extension/tensor/tensor_ptr.h"
+#include "rnexecutorch/data_processing/dsp.h"
+#include "rnexecutorch/data_processing/gzip.h"
+
+ASR::ASR(const models::BaseModel &encoder, const models::BaseModel &decoder,
+         const TokenizerModule &tokenizer)
+    : encoder(encoder), decoder(decoder), tokenizer(tokenizer),
+      startOfTranscriptionToken(
+          this->tokenizer.tokenToId("<|startoftranscript|>")),
+      endOfTranscriptionToken(this->tokenizer.tokenToId("<|endoftext|>")),
+      timestampBeginToken(this->tokenizer.tokenToId("<|0.00|>")) {}
+
+std::vector<int32_t>
+ASR::getInitialSequence(const DecodingOptions &options) const {
+  std::vector<int32_t> seq;
+  seq.push_back(this->startOfTranscriptionToken);
+
+  if (options.language.has_value()) {
+    int32_t langToken =
+        this->tokenizer.tokenToId("<|" + options.language.value() + "|>");
```

Comment:
`std::string` is created again on `+`, maybe create std::string, preallocate memory and then move?
```suggestion
    std::string tokenVal;
    tokenVal.reserve(options.language.value().size() + 4);
    tokenVal.append("<|").append(options.language.value()).append("|>");
    int32_t langToken =
        this->tokenizer.tokenToId(std::move(tokenVal));
```

Reply:
Let's keep it as is

Reply:
Sure, I propose the change because of this: https://clang.llvm.org/extra/clang-tidy/checks/performance/inefficient-string-concatenation.html, so when I will clean the whole codebase and move to clang tidy, this line will probably be changed either way